pymodm/base/options.py

```diff
@@ -86,8 +86,14 @@ def get_field_from_attname(self, attname):
 
     def add_field(self, field_inst):
         """Add or replace a given Field."""
-        orig_field = (self.get_field(field_inst.mongo_name) or
-                      self.get_field_from_attname(field_inst.attname))
+        try:
+            orig_field = self.get_field(field_inst.mongo_name)
+        except Exception:
+            # FieldDoesNotExist, etc. may be raised by subclasses.
+            try:
+                orig_field = self.get_field_from_attname(field_inst.attname)
```

Comment:
Good point! We need this if we want to replace a field with a different mongo_name. Updated the test to demonstrate this.